pkg/actions/lua/lakefs/catalogexport/table_extractor.lua

```diff
@@ -0,0 +1,186 @@
+local pathlib = require("path")
+local strings = require("strings")
+local yaml = require("encoding/yaml")
+local lakefs = require("lakefs")
+local json = require("encoding/json")
+local common = require("lakefs/catalogexport/common")
+
+-- return partition table from path by based on columns in partition_cols
+function get_partition_values(partition_cols, path)
+    local vals = {}
+    splitted_path = strings.split(path, pathlib.default_separator())
+    for _, part in pairs(splitted_path) do
+        for _, col in ipairs(partition_cols) do
+            local prefix = col .. "="
+            if strings.has_prefix(part, prefix) then
+                vals[col] = part:sub(#prefix + 1)
+            end
+        end
+    end
+    return vals
+end
+
+-- extract partition substr from full path
+function extract_partition_prefix_from_path(partition_cols, path)
+    local partitions = get_partition_values(partition_cols, path)
+    local partitions_list = {}
+    -- iterate partition_cols to maintain order  
+    for _, col in pairs(partition_cols) do
+        table.insert(partitions_list, col .. "=" .. partitions[col])
+    end
+    local partition_key = table.concat(partitions_list, pathlib.default_separator())
+    return partition_key
+end
```

Comment:
If I got it right the `extract_partition_prefix_from_path` users `get_parition_values` and the end result is getting the part of the object path based on the paritions / columns.

I assume that the columns are ordered based on the same structure as the object path.
Also, that the number of partitions must match the path information.

Base on the above I suggest the following:

```lua
function extract_partitions_path(partitions, path)
    local idx = 0
    for _, partition in ipairs(partitions) do
        local pattern = partition .. "=[^/]*"
        local i, j = string.find(path, pattern, idx)
        if i == nil then
            return nil
        end
        idx = j + 1
    end
    return string.sub(path, 1, idx)
end
```

It should return the relevant part of the path based on the partitioning used to write the data.

Reply:
Much better thanks! Did almost that (added comment in the code) 👍